web/src/js/ducks/websocket.js

```diff
@@ -1,30 +1,103 @@
+import { ConnectionActions } from '../actions.js'
+import { AppDispatcher } from '../dispatcher.js'
+import * as eventLogActions from './eventLog'
+import * as flowActions from './flows'
+
+const CONNECT = 'WEBSOCKET_CONNECT'
 const CONNECTED = 'WEBSOCKET_CONNECTED'
+const DISCONNECT = 'WEBSOCKET_DISCONNECT'
 const DISCONNECTED = 'WEBSOCKET_DISCONNECTED'
+const ERROR = 'WEBSOCKET_ERROR'
+const MESSAGE = 'WEBSOCKET_MESSAGE'
 
+/* we may want to have an error message attribute here at some point */
+const defaultState = { connected: false, socket: null }
 
-const defaultState = {
-    connected: false,
-    /* we may want to have an error message attribute here at some point */
-}
 export default function reducer(state = defaultState, action) {
     switch (action.type) {
+
+        case CONNECT:
+            return { ...state, socket: action.socket }
```

Comment:
We talked about this before: State should be serializable and immutable (see http://redux.js.org/docs/introduction/ThreePrinciples.html).
What's your argument for moving the WebSocket into state?